grpc/src/main/java/com/linecorp/armeria/server/grpc/HttpJsonTranscodingService.java

```diff
@@ -416,6 +429,53 @@ private static Descriptor findTypeDescriptor(FileDescriptor file, FieldDescripto
         return file.findMessageTypeByName(messageType.getName());
     }
 
+    // to make it more efficient, we calculate whether extract response body one time
+    // if there is no matching toplevel field, we set it to null
+    @Nullable
+    private static String calculateResponseBody(List<FieldDescriptor> topLevelFields, String responseBody) {
+        if (StringUtil.isNullOrEmpty(responseBody)) {
+            return null;
+        }
+        for (FieldDescriptor fieldDescriptor: topLevelFields) {
+            if (fieldDescriptor.getName().equals(responseBody)) {
+                return responseBody;
+            }
+        }
+        return null;
+    }
+
+    @Nullable
+    private static Function<HttpData, HttpData> generateResponseBodyConverter(TranscodingSpec spec) {
+        final String responseBody = spec.httpRule.getResponseBody();
+        if (spec.responseBody == null) {
+            return null;
+        } else {
+            return httpData -> {
+                try (HttpData body = httpData) {
+                    final byte[] array = body.array();
+                    try {
+                        final JsonNode jsonNode = JacksonUtil.readValue(array, JsonNode.class);
+                        final String responseBodyJsonKey =
+                                CaseFormat.UPPER_UNDERSCORE.to(CaseFormat.LOWER_CAMEL, responseBody);
+                        final Iterator<String> fieldNames = jsonNode.fieldNames();
+                        while (fieldNames.hasNext()) {
+                            final String fieldName = fieldNames.next();
+                            if (CaseFormat.UPPER_UNDERSCORE.to(
+                                    CaseFormat.LOWER_CAMEL, responseBody).equalsIgnoreCase(fieldName)) {
+                                final JsonNode responseBodyJsonNode = jsonNode.get(responseBodyJsonKey);
+                                final byte[] bytes = JacksonUtil.writeValueAsBytes(responseBodyJsonNode);
+                                return HttpData.copyOf(bytes);
```

Comment:
```suggestion
                            if (responseBody.equalsIgnoreCase(fieldName)) {
                                final JsonNode responseBodyJsonNode = jsonNode.get(responseBodyJsonKey);
                                final byte[] bytes = JacksonUtil.writeValueAsBytes(responseBodyJsonNode);
                                return HttpData.wrap(bytes);
```

Reply:
The response body string is usually snake case but the json node key is usually in camel case. I think there should be some conversion when iterating the json node.